proc_nutation.py

```diff
@@ -1,53 +1,86 @@
+from pylab import *
 from pyspecdata import *
 from scipy.optimize import minimize
-from proc_scripts import hermitian_function_test,zeroth_order_ph,postproc_dict,fl_mod,center_CPMG_echo
+from proc_scripts import hermitian_function_test,zeroth_order_ph,postproc_dict,fl_mod
 from sympy import symbols
 from numpy import *
 fl = fl_mod()
 t2 = symbols('t2')
 logger = init_logging("info")
-for searchstr,exp_type,nodename,postproc in [
-    ['200219_nutation_alex_probe','test_equip','nutation','spincore_nutation_v1']
+max_kHz = 200
+for searchstr,exp_type,nodename,postproc,freq_slice in [
+    #['201211_Ni_sol_probe_nutation_1','nutation','nutation',
+    #    'spincore_nutation_v1',(-5000,13000)],
+    #['210302_Ni_cap_probe_nutation_1','nutation','nutation',
+    #    'spincore_nutation_v1',(-4e3,1.2e4)]
+    ['210120_Ni_sol_probe_nutation_amp_3','nutation','nutation',
+        'spincore_nutation_v2',(-15.2e3,11.01e3)]
     ]:
     s = find_file(searchstr,exp_type=exp_type,expno=nodename,postproc=postproc,
             lookup=postproc_dict) 
- 
+    print(s.dimlabels)
+    if 'amp' in s.dimlabels:
+        plen = s.get_prop('acq_params')['p90_us']*1e-6
+        logger.info(strm('pulse length is:',plen))
     # {{{ do the rough centering before anything else!
     # in particular -- if you don't do this before convolution, the
     # convolution doesn't work properly!
-    s.ft(['ph2','ph1'])
-    # }}}
-    
+    s.ift('t2')
     # {{{ centering of data using hermitian function test
-    best_shift = hermitian_function_test(s['ph2',0]['ph1',1])
-    logger.info(strm("best shift is",best_shift))
-    s.setaxis('t2', lambda x: x-best_shift).register_axis({'t2':0})
+    if 'p90' in s.dimlabels:
+        best_shift = hermitian_function_test(s['ph2',0]['ph1',1])
+        logger.info(strm("best shift is",best_shift))
+        s.setaxis('t2', lambda x: x-best_shift).register_axis({'t2':0})
+        fl.next('with time shift')
+        fl.image(s)
+        fl.next('freq domain after time correction')
+        s.ft('t2')
+        fl.image(s)
+    #}}}
+    if 'amp' in s.dimlabels:
+        s.setaxis('t2', lambda x: x-abs(s['ph1',1]['ph2',-2]).mean_all_but('t2').argmax('t2').item())
     #}}}
-    
-    s.ft('t2',pad=4096)
-    
+    fl.next('t domain centered')
+    fl.image(s)
+    fl.next('freq domain centered')
+    s.ft('t2')
+    fl.image(s)
+    s.ift('t2')
     #{{{ selecting coherence and convolving
     s = s['ph2',0]['ph1',1]
-    fl.next('select $\\Delta p$ and convolve')
-    s.convolve('t2',50)
+    fl.next('select $\\Delta p$')
+    if 'p90' in s.dimlabels:
+        s.convolve('t2',50)
```

Comment:
it's weird that this convolution depends on whether it's variable amplitude vs. variable pulse length

Reply:
no more convolution